docs/data/indexers/README.mdx

```diff
@@ -1,14 +1,113 @@
+
+## What exactly is indexing?
+
+The word "indexing" has come to encompass a large array of use-cases. What all of them have in common is that they process and structure blockchain data. Let's think of it in the order of what you, an app-builder or small team, will need:
+
+### 1. Off-the-shelf APIs for common data {#portfolio-apis}
+
+This is the kind of stuff you might get from Horizon today, but can include more than that. For example, many indexing services in this category also provide the "every time an NFT changed ownership" kind of data mentioned above.
+
+Lots of apps need this kind of data, and it tends to have a standard shape. This makes it profitable for companies to build SaaS products, offering API access for a price (often with a free tier, appropriate for your early-stage startup or hackathon project).
+
+In the biz, these are called _Portfolio APIs._ Well-known companies offering them:
+
+- [Alchemy]: most popular choice on Ethereum; now in talks to expand their service to Stellar, with a targeted launch in the first half of 2026.
+
+- [Allium]: currently building Stellar support, launching Q1 2026
+
+- [OBSRVR]: a home-grown Stellar-native offering, providing both RPC services and [Obsrvr Gateway](https://www.withobsrvr.com/products/gateway), promising "powerful APIs for real-time data fetching, transaction processing, and easy integration into existing systems."
+
+- [Horizon]: as mentioned above, Horizon will soon be deprecated, but for now it is the only way to get some of this data.
+
+### 2. Streaming & transforming data to a custom app database {#custom-transformations}
+
+Not all apps have data that fits a standard shape. Most, probably, need custom data transformation.
+
+Consider again the NFT example. Wouldn't it be great to return all needed NFT data in _one_ request, rather than N+1? That's what this style of indexing provides. And even better: you can (often) use these indexing solutions to aggregate data from multiple sources. If your NFT app stores additional information on IPFS, you could consolidate all of that off-chain data to your database as well. This makes your app much faster, saves you network & RPC requests, and creates new architecture possibilities.
+
+Well-known options:
+
+- [The Graph]: one of the earliest and most popular-at-the-time options on Ethereum. They now offer three main products:
+  1. [Subgraphs](https://thegraph.com/docs/en/subgraphs/developing/subgraphs/): their main & most famous offering, providing a [decentralized approach](https://thegraph.com/docs/en/subgraphs/developing/publishing/publishing-a-subgraph/) to custom data transformation & hosting. The Graph's use of GraphQL APIs made GraphQL a popular choice for this entire category of indexing.
+  2. **Token API**, for Indexing Use Case #1 described above
+  3. **Substreams**, for Use Case #3 described below
```

Comment:
would be helpful to add notes or links to the particular sections above/below you're referring to.